File: ePrijevozSarajevo/UI/eprijevoz_mobile/lib/models/rezervacija_prostora20022025.dart
import 'package:json_annotation/json_annotation.dart';

part 'rezervacija_prostora20022025.g.dart';

@JsonSerializable()
class RezervacijaProstora20022025 {
  int? rezervacijaProstora20022025Id;
  int? userId;
  DateTime? pocetakRezervacije;
  DateTime? trajanjeRezervacije;
  String? napomena;
  int? radniProstorId;
  int? statusRezervacijeId;

  RezervacijaProstora20022025(
      {this.rezervacijaProstora20022025Id,
      this.userId,
      this.pocetakRezervacije,
      this.trajanjeRezervacije,
      this.napomena,
      this.radniProstorId,
      this.statusRezervacijeId});

  factory RezervacijaProstora20022025.fromJson(Map<String, dynamic> json) =>
      _$RezervacijaProstora20022025FromJson(json);

  Map<String, dynamic> toJson() => _$RezervacijaProstora20022025ToJson(this);
}

File: ePrijevozSarajevo/ePrijevozSarajevo.Model/RezervacijaProstora20022025.cs
namespace ePrijevozSarajevo.Model
{
    public class RezervacijaProstora20022025
    {
        public int RezervacijaProstora20022025Id { get; set; }

        public int UserId { get; set; }
        public DateTime PocetakRezervacije { get; set; }
        public DateTime TrajanjeRezervacije { get; set; }
        public string Napomena { get; set; } = null!;

        public int RadniProstorId { get; set; }

        public int StatusRezervacijeId { get; set; }
    }
}
